pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch_test.go

```diff
@@ -802,8 +812,9 @@ func BenchmarkTelemetryV2Filters(b *testing.B) {
 	var got interface{}
 	b.ResetTimer()
 	for n := 0; n < b.N; n++ {
+		copied := proto.Clone(l)
```

Comment:
why did you have to add this? This is suspicious, are we modifying the EnvoyFitlter somewhere? I don't think we should be doing that, caused a lot of problems doing a similar thing with VS

Reply:
This is a bug fix, as i tested, i found that the running time is grown as the times increase. It is because, the listener length is growing each time.

Reply:
This can be seen as the memory leaks

Reply:
I am pretty sure this is a bug in listener patch code. We shouldn't be modifying the listeners we pass in. Otherwise we will run into issues. For example, if we have some filter as constant and we modify that, its going to be very bad

Reply:
Or maybe its not an issue... what do you think?